### PR TITLE
:sparkles: add clusterclass

### PR DESCRIPTION
**Pull request by farodin91**
**What is the purpose of this pull request/Why do we need it?**

We use clusterclass to automate updates and improve cluster handling.

**Description of changes:**

Add support for clusterclass

**Checklist:**
- [x] Includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)

### CODE REVIEW

File: templates/clusterclass-template.yaml [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+  - definitions:
+    - jsonPatches:
+      - op: add
+        path: /spec/template/spec/kubeadmConfigSpec/users
+        valueFrom:
+          template: |
+            - name: capv
+              sshAuthorizedKeys:
+              - '{{ .sshKey }}'
+              sudo: ALL=(ALL) NOPASSWD:ALL
```
Comment (lubedacht) [inline templates/clusterclass-template.yaml:46]: In our base template the user is root. Also applies to the other patches in the file

File: templates/cluster-template-topology.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: "${CLUSTER_NAME}-credentials"
+  namespace: '${NAMESPACE}'
+type: Opaque
+stringData:
+  token: "${IONOS_TOKEN}"
+  apiURL: "${IONOS_API_URL:-https://api.ionos.com/cloudapi/v6}"
```
Comment (lubedacht) [inline templates/cluster-template-topology.yaml:45]: We aim to have `newlines` at the end of files.
Usually in your IDE you have a setting to append newlines when you save a file. I can recommend that setting.

File: templates/cluster-template-topology.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  name: '${CLUSTER_NAME}'
+  namespace: '${NAMESPACE}'
+spec:
+  topology:
+    class: '${CLUSTER_CLASS_NAME}'
+    controlPlane:
+      replicas: ${CONTROL_PLANE_MACHINE_COUNT}
+    variables:
+    - name: sshKey
+      value: '${VSPHERE_SSH_AUTHORIZED_KEY}'
```
Comment (lubedacht) [inline templates/cluster-template-topology.yaml:16]: copy pasta.
Also we can provide multiple ssh keys with `${IONOSCLOUD_MACHINE_SSH_KEYS}`. So you might want to use the plural form for the variable

File: templates/cluster-template-topology.yaml (hunk 2 above)
Comment (farodin91) [reply templates/cluster-template-topology.yaml:45]: I tend to work with editorconfig everywhere.

**Comment by farodin91** [comment]
Replaced by #141

**Pull request closed without merging**